src/plugins/intel_gpu/src/runtime/kernels_cache.cpp

```diff
@@ -131,10 +131,48 @@ void kernels_cache::get_program_source(const kernels_code& kernels_source_code,
                 current_bucket.push_back(batch_program(bucket_id, batch_id, options, batch_header_str));
             }
 
+            auto get_base_kernel_name = [](const std::string& unique_kernel_name) -> std::string {
+                int matched = 0;
+                size_t pos = unique_kernel_name.length();
+                if (unique_kernel_name.substr(pos-4, 4).compare("__sa") == 0) {
+                    pos -= 4;
+                }
+                while (pos > 0) {
+                    pos -= 1;
+                    if (unique_kernel_name.at(pos) == '_') {
+                        matched += 1;
+                    }
+                    if (matched == 3) {
+                        break;
+                    }
+                }
+                return unique_kernel_name.substr(0, pos);
+            };
+
+            // This is a temporary walk-around to avoid severe performance drop.
+            // It will be removed after OpenCL compiler is updated.
+            auto need_separate_batch = [&](std::string& unique_kernel_name) -> bool {
+                const std::vector<std::string> special_kernels = {"gemm_tiled_opt"};
+
+                // check if the current kernel name is in special_kernels
+                auto target_base_kernel_name = get_base_kernel_name(entry_point);
+                if (std::count(special_kernels.begin(), special_kernels.end(), target_base_kernel_name) > 0)
+                    return true;
```

Comment:
If currerent entryu has gemm_tiled_opt => it will need_seperate_batch : Is this the intention? 
(Current behavior seems so) 
If it is so, why not just simply check : 
 if (entry_point.find("gemm_tiled_opt") != string::npos)
    => need_separate_batch?

Reply:
Oh! I updated it as you reviewed. Thank you!